### ISSUE TITLE
[BUG] Custom rule not triggered for uploaded binary.

### ISSUE THREAD

**Issue by sshveta** [konveyor/analyzer-lsp#1094]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Konveyor version

upstream

### Priority

Critical

### Current Behavior

Run analysis on [binary](https://github.com/konveyor/tackle2-ui/blob/main/cypress/fixtures/jee-example-app-1.0.0.ear) 

with custom rule 
================

```
- category: optional
  customVariables: []
  description: ServletAuthentication CUSTOM RULE - ServletAuthentication
  effort: 1
  labels:
    - konveyor.io/source=springboot
    - konveyor.io/source=weblogic
    - konveyor.io/target=eap7
    - konveyor.io/target=eap8
    - hibernate
    - configuration
  links:
    - title: This link should take you to the Konveyor main page
      url: https://www.konveyor.io/
    - title: This link should take you to the Konveyor organization GitHub page
      url: https://github.com/konveyor
  message:
    This is a custom rule for testing that should trigger issues in imports
    of ServletAuthentication
  ruleID: basic-custom-rule
  when:
    java.referenced:
      location: IMPORT
      pattern: weblogic.servlet.security.ServletAuthentication
```


### Expected Behavior

Analysis:                                                                                                                      
                                                                                                                                 
  1. The .ear file contains the reference:                                                                                       
  jee-example-app-1.0.0.ear                                                                                                      
  └── jee-example-web.war                                                                                                        
      └── WEB-INF/classes/com/acme/anvil/AuthenticateFilter.class                                                                
  2. The AuthenticateFilter.class bytecode contains:
    - Class reference: weblogic/servlet/security/ServletAuthentication
    - Method call: ServletAuthentication.invalidateAll()
  3. The custom rule pattern matches:
  when:
    java.referenced:
      location: IMPORT
      pattern: weblogic.servlet.security.ServletAuthentication

### How Reproducible

Always (Default)

### Steps To Reproduce

         source: "Upload a local binary",
        target: ["Application server migration to"],
        binary: ["jee-example-app-1.0.0.ear"],
        customRule: ["basic-custom-rule.yaml"],

### Environment

```markdown
- **Analyzer Image SHA** 
- **JDTLS  base image SHA**
- **Provider Settings JSON**
```

### Anything else?

_No response_

**Comment by eemcmullan** [comment]
@sshveta going to close this issue, as this binary is not valid. It is failing to decompile with `ERRO[0015] failed to get list of dependencies for provider  error="binary is not yet resolved"` as there are bad `.jars` that we cannot decompile. Before, we would skip these issues. But now, we fail analysis when this occurs. If you notice this issue with other binaries, please re-open.
